actioncable/lib/action_cable/server/configuration.rb

```diff
@@ -28,7 +29,12 @@ def channel_paths
 
       def channel_class_names
         @channel_class_names ||= channel_paths.collect do |channel_path|
-          Pathname.new(channel_path).basename.to_s.split('.').first.camelize
+          dirname, basename = Pathname.new(channel_path).split
+
+          [
+            dirname.each_filename.to_a.split{ |f| f === 'channels' }.last,
```

Comment:
> Why do we need "channels" part of path?

To get namespaces inside `channels` directory.
Yes, you're right, You should add `lib/channels` to `channel_load_paths`.

```ruby
ActionCable.server.config.channel_load_paths << Rails.root.join('lib/channels')
#=> [#<Pathname:/rails-test/app/channels>, #<Pathname:/rails-test/lib/channels>] 
```